tasks/preflight.yml

```diff
@@ -51,14 +56,41 @@
   set_fact:
     cmos_exporter_systemd_version: "{{ __systemd_version.stdout_lines[0] | regex_replace('^systemd\\s(\\d+).*$', '\\1') }}"
 
+- block:
+    - name: Get the Couchbase Cluster Info
+      no_log: true
+      uri:
+        url: http://{{ inventory_hostname }}:8091/pools/default
+        method: GET
+        user: "{{ couchbase_username }}"
+        password: "{{ couchbase_password }}"
+        timeout: 2
+      register: _results
+
+    - name: Set the Couchbase Server Version from the result of API call
+      set_fact:
+        couchbase_server_version: "{{ _results.json | community.general.json_query('nodes[?thisNode].version | [0]') }}"
+      when: not couchbase_server_version
+
+    - name: Couchbase Server Version
+      debug:
+        msg: "{{ couchbase_server_version }}"
+
+    - name: Fail when the installed version of Couchbase is 7.x or greater
+      fail:
+        msg: >
+          The CMOS Couchbase Exporter does not need to be installed when using Couchbase 7.x+ metrics are natively exposed through a /metrics endpoint.
+          See: https://docs.couchbase.com/server/current/manage/monitor/set-up-prometheus-for-monitoring.html
+      when: (couchbase_server_version | split('.') | first | int) >= 7
```

Comment:
Could this be `couchbase_server_version is version('7.0.0', '>=')`? https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#comparing-versions

Reply:
Yes that should work, updated.